src/script/cpp_api/s_base.cpp

```diff
@@ -195,29 +200,27 @@ void ScriptApiBase::clientOpenLibs(lua_State *L)
 }
 #endif
 
+#define CHECK(ridx, name) do { \
+	lua_rawgeti(L, LUA_REGISTRYINDEX, ridx); \
+	FATAL_ERROR_IF(lua_type(L, -1) != LUA_TFUNCTION, "missing " name); \
+	lua_pop(L, 1); \
+	} while (0)
+
 void ScriptApiBase::checkSetByBuiltin()
 {
 	lua_State *L = getStack();
 
 	if (m_gamedef) {
-		lua_rawgeti(L, LUA_REGISTRYINDEX, CUSTOM_RIDX_READ_VECTOR);
-		FATAL_ERROR_IF(lua_type(L, -1) != LUA_TFUNCTION, "missing read_vector");
-		lua_pop(L, 1);
-
-		lua_rawgeti(L, LUA_REGISTRYINDEX, CUSTOM_RIDX_PUSH_VECTOR);
-		FATAL_ERROR_IF(lua_type(L, -1) != LUA_TFUNCTION, "missing push_vector");
-		lua_pop(L, 1);
-
-		lua_rawgeti(L, LUA_REGISTRYINDEX, CUSTOM_RIDX_READ_NODE);
-		FATAL_ERROR_IF(lua_type(L, -1) != LUA_TFUNCTION, "missing read_node");
-		lua_pop(L, 1);
-
-		lua_rawgeti(L, LUA_REGISTRYINDEX, CUSTOM_RIDX_PUSH_NODE);
-		FATAL_ERROR_IF(lua_type(L, -1) != LUA_TFUNCTION, "missing push_node");
-		lua_pop(L, 1);
+		CHECK(CUSTOM_RIDX_READ_VECTOR, "read_vector");
+		CHECK(CUSTOM_RIDX_PUSH_VECTOR, "push_vector");
+		CHECK(CUSTOM_RIDX_READ_NODE, "read_node");
+		CHECK(CUSTOM_RIDX_PUSH_NODE, "push_node");
+		CHECK(CUSTOM_RIDX_PUSH_MOVERESULT1, "push_moveresult1");
```

Comment:
```C++
2024-04-30 20:35:59: ERROR[Main]: In thread 7ffff6227940:
2024-04-30 20:35:59: ERROR[Main]: /data/Games/Minetest/minetest/src/script/cpp_api/s_base.cpp:db: void ScriptApiBase::checkSetByBuiltin(): A fatal error occurred: missing push_moveresult1

Thread 1 "minetest" received signal SIGABRT, Aborted.
__pthread_kill_implementation (no_tid=0, signo=6, threadid=140737322842432) at ./nptl/pthread_kill.c:44
44	./nptl/pthread_kill.c: Datei oder Verzeichnis nicht gefunden.
(gdb) bt
#0  __pthread_kill_implementation (no_tid=0, signo=6, threadid=140737322842432) at ./nptl/pthread_kill.c:44
#1  __pthread_kill_internal (signo=6, threadid=140737322842432) at ./nptl/pthread_kill.c:78
#2  __GI___pthread_kill (threadid=140737322842432, signo=signo@entry=6) at ./nptl/pthread_kill.c:89
#3  0x00007ffff721b476 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#4  0x00007ffff72017f3 in __GI_abort () at ./stdlib/abort.c:79
#5  0x000055555566d0be in fatal_error_fn(char const*, char const*, unsigned int, char const*) [clone .cold] ()
#6  0x00005555559d5a31 in ScriptApiBase::checkSetByBuiltin() ()
#7  0x00005555556b63a4 in Client::loadMods() ()
#8  0x00005555556b21ea in Client::ReceiveAll() ()
...
```
* The lambda is run 3 times on a local server: 1x SSM + 1x server emerge + 1x server async

--> `misc_s.lua` is not run for CSM, thus the the RDIX. See backtrace.

Reply:
fixed.